### ISSUE TITLE
Generator stars count has gone to zero

### ISSUE THREAD

**Issue by gigabo** [redfin/react-server#776]
![image](https://cloud.githubusercontent.com/assets/115908/20777482/2afe8662-b71c-11e6-94f3-bfc9752ecace.png)

@doug-wade Any ideas?  [This was correct](http://yeoman.io/generators/) for a while, at least. 😕 

Also, is the other generator there the same thing under a different name?

**Comment by mischah** [comment]
We had serious problems after having more than 5000 generator in that list.

The generator list is totally new and shining and should be merged in master soon. See yeoman/yeoman-generator-list#52
 :sparkles:

**Comment by mischah** [comment]
More insights:

We are using the npms.io API to fetch generators from.

Having a look at https://api.npms.io/v2/package/generator-react-server shows that the result is missing the `github` object.

You find an answer about how to fix this over here: https://github.com/npms-io/npms-analyzer/issues/112

**Comment by gigabo** [comment]
Hi @mischah thanks for looking at this!

What is the `github` object that it's looking for?  Is it separate from the `repository` field?

I tried using `hosted-git-info` as suggested in npms-io/npms-analyzer#112.  It _seems_ to work:

```
> require("hosted-git-info").fromUrl("redfin/react-server")
{ type: 'github',
  protocols: [ 'git', 'http', 'git+ssh', 'git+https', 'ssh', 'https' ],
  domain: 'github.com',
  treepath: 'tree',
  filetemplate: 'https://{auth@}raw.githubusercontent.com/{user}/{project}/{committish}/{path}',
  bugstemplate: 'https://{domain}/{user}/{project}/issues',
  gittemplate: 'git://{auth@}{domain}/{user}/{project}.git{#committish}',
  sshtemplate: 'git@{domain}:{user}/{project}.git{#committish}',
  sshurltemplate: 'git+ssh://git@{domain}/{user}/{project}.git{#committish}',
  browsetemplate: 'https://{domain}/{user}/{project}{/tree/committish}',
  docstemplate: 'https://{domain}/{user}/{project}{/tree/committish}#readme',
  httpstemplate: 'git+https://{auth@}{domain}/{user}/{project}.git{#committish}',
  shortcuttemplate: '{type}:{user}/{project}{#committish}',
  pathtemplate: '{user}/{project}{#committish}',
  pathmatch: /^[\/]([^\/]+)[\/]([^\/]+?)(?:[.]git)?$/,
  protocols_re: /^(git|http|git\+ssh|git\+https|ssh|https):$/,
  user: 'redfin',
  auth: null,
  project: 'react-server',
  committish: null,
  default: 'shortcut' }
```

**Comment by mischah** [comment]
Hej @gigabo, 

the `github` object should be in the JSON response from the npms.io API.

Compare the following two responses.

+ Example Yeoman generator which has the `github` object
  + https://api.npms.io/v2/package/generator-http-fake-backend
+ The React server Yeoman generator
  + https://api.npms.io/v2/package/generator-react-server

The `github` object has a property called `starsCount` among others.

I have no idea about the reason for the absence of the `github` object in your case. Because the [package.json](https://github.com/redfin/react-server/blob/master/packages/generator-react-server/package.json) of your Yeoman generator references to this Github repo:

```json
"repository": "redfin/react-server"
```

Maybe it has something to to with your unusual setup having different npm packages in this Github repository in the [packages](https://github.com/redfin/react-server/tree/master/packages) subdirectory?!?

So I would suggest opening an issue over here: https://github.com/npms-io/npms-analyzer/issues

**Comment by mischah** [comment]
Oops. This is a known issue with so called »monorepos«.
See https://github.com/npms-io/npms-analyzer/issues/83

So your setup doesn’t seem to be that unusual 😳

**Comment by gigabo** [comment]
Oh wow.  That's a bummer.

Thanks for tracking that down @mischah.

**Comment by mischah** [comment]
Yeah. Going to zero stars in the generator list really is a bummer 😖 

I just left a message over here: npms-io/npms-analyzer#83 :disappointed:
